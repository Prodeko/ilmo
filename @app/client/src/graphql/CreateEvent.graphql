fragment CreatedEvent on Event {
  id
  name
  ownerOrganization {
    id
    name
  }
}

mutation CreateEvent($name: String!, $desc: String!, $org_id: UUID!) {
  createEvent(
    input: {
      event: { name: $name, description: $desc, ownerOrganizationId: $org_id }
    }
  ) {
    event {
      id
      ...CreatedEvent
    }
  }
}
