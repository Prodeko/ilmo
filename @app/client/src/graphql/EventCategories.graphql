query EventCategories {
  ...EventCategories_Query
}
