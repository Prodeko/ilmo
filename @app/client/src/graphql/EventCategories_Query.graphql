fragment EventCategories_EventCategory on EventCategory {
  id
  name
  description
  isPublic
}

fragment EventCategories_Query on Query {
  ...SharedLayout_Query
  eventCategories {
    nodes {
      id
      ...EventCategories_EventCategory
    }
  }
}
