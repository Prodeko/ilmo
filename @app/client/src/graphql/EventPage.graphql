query EventPage($id: UUID!) {
  ...EventPage_Query
}
