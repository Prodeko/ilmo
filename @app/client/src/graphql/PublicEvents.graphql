query PublicEvents {
  publicEvents {
    categoryName
    id
    name
    ownerOrganizationName
    startTime
    isPublic
  }
}
