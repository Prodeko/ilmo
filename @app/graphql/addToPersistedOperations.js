const map = require("./server.json");
const { promises: fsp } = require("fs");

async function main() {
  await Promise.all(
    Object.entries(map).map(([hash, query]) =>
      fsp.writeFile(`${__dirname}/.persisted_operations/${hash}.graphql`, query)
    )
  );
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});
