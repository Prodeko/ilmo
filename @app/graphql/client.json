{
   "RenderEmailTemplate": "d86f0b25c02f929d08411b0d58597c7115575d51df09d8e5c743e254f44d3c5e",
   "RenderEmailTemplates": "b7093c4a6d385e4a30084683a73f3074f0eaed478b3d9f58ceebef12145464fd",
   "CreateEventCategoryPage": "03f7010d3c96389f5598d3b93fd15d7b5c8a3a2ac3a0f9c5741a6460a428a860",
   "CreateEventCategory": "b5cfbc37e1ac0b2997177dc1caa52944f13d41035e130e3ada698f5b22a15e9f",
   "DeleteEventCategory": "f555d3fc31063758bc496e6fca37600cb6734d08aa6a8f9337f1073636a1bb8d",
   "ListEventCategories": "ade3b3b7163163f82ac01db9d382e8ec4171c5e3258c02ae6adc91e12785170a",
   "UpdateEventCategoryPage": "754248497408041290587849a820fc2c2c9f9bc7ebaa140a375506b3647d3536",
   "UpdateEventCategory": "3bf84e660c99a57e93c875c71fe468f06ddc151afc6c43cd9e5983e2764e81b2",
   "CreateEventPage": "54923f42e1ba031b33688e6581211104ea4ae057ad3d40171a14152d1cdc764a",
   "CreateEvent": "2263eb525bfe274c5f2af9a98f3bd6edc499f9a0e14a35dcb2d2a453daf75941",
   "DeleteEvent": "75c86da8b31d18d79c4dcab26bacfffb860b4c7933568ca830731c793f815338",
   "ListEvents": "5d424ea1e4f830e49e164d3f360b19f46eca083310030f921c9a2100a47e2ea1",
   "UpdateEventPage": "018ef2d5fd12b5193ce1e4250972a0913a37353d060f6f73fc9b5dc2036a59b1",
   "UpdateEvent": "9f8c41ac01dd9ed51e5e15e69e9f22f61594e816fddba0ac52852cff0a601fc7",
   "InviteToOrganization": "c639d16546c2d6c7133350ef65627885f7e11a1e33ec9ae099d0bb4a9b9af331",
   "OrganizationBySlug": "7d4e6fe382bf3e0467c9372406e18fd1e63b6aa3f1aea2530a66c60cdfa3da2d",
   "OrganizationPage": "4e233dde8224c654c3a46223975e5be5d814887e12a96e3f356d55823205c4f9",
   "RemoveFromOrganization": "234a33748153a85850a5e8cd76126a46c10f300f3e3deee0b1e8b832aa66948d",
   "TransferOrganizationOwnership": "949e86df3cd114d207d517c292df1fc19b09e8ce9609c4ae243b61f002959bd7",
   "UpdateOrganization": "e7fa7591569dc63a63e328a8aa8c862ddb316ea999e80bd7009500cd2d1ae2ac",
   "EventPage": "1d1d544cf4d88be73955f8a4b092dc70f17ae90ed82bb986157744aaa5a20e3e",
   "EventRegistrations": "b1fa2d07d27a732d287486071d0baece76ddfd7c1c5872d108b578992645b5d9",
   "HomePageEvents": "d9f898c72eab71ac1cee2557937b41834434a63fe0d9dd794f980ae9abc46b14",
   "HomePage": "7bd6d19290c24e7302c029d1dccd5ff425f1c1ffc83a8a3b49be35a024ebbc64",
   "Shared": "3d8f3a22d33c3600bc2d5bde4c561b34cfa12a9240f603791d3aa27e1290c503",
   "AcceptOrganizationInvite": "29b135f119928bfc6de24b8aa6ec513ed62cb3cadf95eec1da607a3c760df7e9",
   "CreateOrganization": "64afdf0107684997a54cccc615c73068491d296ea312500540c2d341e978a9b1",
   "DeleteOrganization": "39f623b7dd9b3111922c0937f5985ef32d8de76b88898b043daaf53dc20877b2",
   "InvitationDetail": "457f0ee2e793f1e5a1cbf546c6b72654ad6f976419e8b84fda8b5ade21764f05",
   "CreateEventQuotas": "2c54237d129d3f7ebc40e672b4155f6956fd39ca0953f76a437cae43c138d4f8",
   "UpdateEventQuotas": "8c81526ed4dfe2178d32b3ba236238d12208fc33e24303b63049630f8dab15b9",
   "ClaimRegistrationToken": "4c3ed5fa6081518ea8090afeea30003d6670336cbc221ebf8c4fcb4bba7e98ad",
   "CreateEventRegistration": "198750facb7638ad82861475ce68890a9a8d29f7844461ee07086d225f43f3a5",
   "DeleteEventRegistration": "be0f72e306dcf5e22dbaa962b5228422a5a0bd12dc68d688ee8b9a28b479318b",
   "EventRegistrationPage": "92a1fbc7c1528843416a32dd15e85cc1568ecfdf1404692e12046908c923b67e",
   "UpdateEventRegistrationPage": "d4b5e5ce523d195cd9dcdfa9217cb5b7fafdc00af5d25a2b7a4d332af5359ec4",
   "UpdateEventRegistration": "b24f20c7b3ec779ac9f462c4a7aa42a4ed01b758dfc6f2e0ec8ddef7fd06f748",
   "AddEmail": "b1951baa0d8c160e92162eabfb6fa4202c83502fe9417288358a932226d14ad2",
   "ChangePassword": "2532c897bead71cb8b9171c240298a47c4b6b13197886c990ccfcc937356ceab",
   "ConfirmAccountDeletion": "28c6757c5282d5f04e93b027aea1642ffd3afc54ea99778983d476e615919c04",
   "CurrentUserAuthentications": "acb7e4d5dac1d3a24b449a997464e569d00374f4c2a28d0ed6153271b458e0b8",
   "CurrentUserUpdated": "5305aa1351d8d66402f6350dbd9ca5dca7f51bdf3912b57c0c5178cc188f6597",
   "DeleteEmail": "d01ef88ad9ca62a5fde36629c391e5857b46e0f239cb9a731398a78c1f2f2310",
   "ForgotPassword": "6234478860eb2415e3714f597e11b7fc16719c4b1d449e03f8806bd8f44312f3",
   "Login": "07b9497235afa4016a486e034a9d91c9cb9c322e91393cec998f25fa6e98426a",
   "Logout": "83b38cbc6a69d6bab5320022c9afa11c0bb734c2998e00c75444421dc874719d",
   "MakeEmailPrimary": "585b5baf970ccdcfc9031b3063524111fb24c972b41b6902e0b97c26eed8407d",
   "Register": "dcbde1d565e29bfd0f92394c44c1343ad70b736f402cdbdae0c5e51b7266be09",
   "RequestAccountDeletion": "a15c749947f08c4c76e36aaa48042537e7f011318cb0424a67cceb336b8f1f7d",
   "ResendEmailVerification": "91c3f0e452f0295520ee02ffbb79759b6a96cac6fc9ccaa514ffaa4e1ca8deba",
   "ResetPassword": "0c6997c5a3b8b971a99c59a3ec6bd05d7fd24ff1aa3d7bbcc8a4bb0808de0ad7",
   "SettingsEmails": "ba89b6c205928e16cfa0e8a4364f2433af4ff55661eb62066d1c05b4ad00a970",
   "SettingsPassword": "4ccd52e00c526690ac5b44ca911534e96f936acd4ff02a2785d9932bbe11ff6c",
   "SettingsProfile": "64623360e96bb37149d327c854482fef1b638a62e9ada5f8cadc13e6b28354a4",
   "UnlinkUserAuthentication": "efef5d154b91120114a5ae97193bb83daee1223c150c8bda2d6655a47482fb70",
   "UpdateUser": "f3860479c8aaf60b1901dca62029b57c96953064a5a65a4ed92cf8e5173ed441",
   "VerifyEmail": "37dc92ddb2ef943241e08bdfa7ee71c207fc876df739992d93ded6b9e76819d1"
}