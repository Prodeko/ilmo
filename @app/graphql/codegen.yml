overwrite: true
schema: "../../data/schema.graphql"
documents: "../client/src/**/*.graphql"
hooks:
  afterAllFileWrite:
    - node addToPersistedOperations.js
config:
  avoidOptionals:
    field: true
    inputValue: false
    object: false
  scalars:
    Datetime: "string"
    JSON: "{ [key: string]: any }"
  noGraphQLTag: false
  withHOC: false
  withComponent: false
  withHooks: true
  reactApolloVersion: 3
  preResolveTypes: true
generates:
  index.tsx:
    plugins:
      - add:
          content: "/* DO NOT EDIT! This file is auto-generated by graphql-code-generator - see `codegen.yml` */"
      - "typescript"
      - "typescript-operations"
      - "typescript-react-apollo"
  client.json:
    plugins:
      - graphql-codegen-persisted-query-ids:
          output: client
          algorithm: sha256
  server.json:
    plugins:
      - graphql-codegen-persisted-query-ids:
          output: server
          algorithm: sha256
