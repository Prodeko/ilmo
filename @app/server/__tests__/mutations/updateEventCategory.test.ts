import { UpdateEventCategoryDocument } from "../../../graphql"
import {
  asRoot,
  createEventDataAndLogin,
  deleteTestData,
  runGraphQLQuery,
  setup,
  teardown,
} from "../helpers"

beforeEach(deleteTestData)
beforeAll(setup)
afterAll(teardown)

describe("UpdateEventCategory", () => {
  it("can update an existing event category", async () => {
    const { organization, eventCategory, session } =
      await createEventDataAndLogin()

    await runGraphQLQuery(
      UpdateEventCategoryDocument,

      // GraphQL variables:
      {
        input: {
          id: eventCategory.id,
          patch: {
            name: {
              fi: "P채ivitetty testikategoria",
              en: "Updated test event category",
            },
            description: {
              fi: "P채ivitetty testikuvaus",
              en: "Updated test description",
            },
            ownerOrganizationId: organization.id,
          },
        },
      },

      // Additional props to add to `req` (e.g. `user: {sessionId: '...'}`)
      {
        user: { sessionId: session.uuid },
      },

      // This function runs all your test assertions:
      async (json, { pgClient }) => {
        expect(json.errors).toBeFalsy()
        expect(json.data).toBeTruthy()

        const updatedEventCategory =
          json.data!.updateEventCategory.eventCategory

        const { rows } = await asRoot(pgClient, () =>
          pgClient.query(
            `SELECT * FROM app_public.event_categories WHERE id = $1`,
            [updatedEventCategory.id]
          )
        )

        if (rows.length !== 1) {
          throw new Error("Event category not found!")
        }
        expect(rows[0].id).toEqual(updatedEventCategory.id)
        expect(rows[0].name.fi).toEqual("P채ivitetty testikategoria")
      }
    )
  })

  it("can't update an event category while logged out (RLS policy)", async () => {
    const { eventCategory } = await createEventDataAndLogin({
      registrationOptions: { create: false },
    })
    await runGraphQLQuery(
      UpdateEventCategoryDocument,

      // GraphQL variables:
      {
        input: {
          id: eventCategory.id,
          patch: {
            description: {
              fi: "P채ivitetty testikuvaus",
              en: "Updated test description",
            },
          },
        },
      },

      // Additional props to add to `req` (e.g. `user: {sessionId: '...'}`)
      {},

      // This function runs all your test assertions:
      async (json) => {
        expect(json.errors).toBeTruthy()

        const message = json.errors![0].message
        expect(message).toEqual(
          "No values were updated in collection 'event_categories' because no values you can update were found matching these criteria."
        )
      }
    )
  })
})
