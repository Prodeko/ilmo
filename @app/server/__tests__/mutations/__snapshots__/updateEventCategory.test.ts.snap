// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateEventCategory can update an existing event category as admin 1`] = `
Object {
  "data": Object {
    "updateEventCategory": Object {
      "eventCategory": Object {
        "color": "[random-3]",
        "description": "[random-1]",
        "id": "[id-1]",
        "name": "[random-2]",
        "ownerOrganization": Object {
          "color": "[random-6]",
          "id": "[id-2]",
          "name": "[random-4]",
          "slug": "[random-5]",
        },
        "updatedAt": "[timestamp-1]",
      },
    },
  },
}
`;

exports[`UpdateEventCategory can't update an event category while logged in as non admin 1`] = `
Object {
  "data": Object {
    "updateEventCategory": null,
  },
  "errors": Array [
    Object {
      "extensions": Object {
        "exception": Object {},
      },
      "locations": Array [
        Object {
          "column": 3,
          "line": 2,
        },
      ],
      "message": "No values were updated in collection 'event_categories' because no values you can update were found matching these criteria.",
      "path": Array [
        "updateEventCategory",
      ],
    },
  ],
}
`;
